src/devices/Am2320/Am2320.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Device.Model;
+using System.Threading;
+using UnitsNet;
+
+namespace Iot.Device.Am2320
+{
+    /// <summary>
+    /// AM2320 - Temperature and Humidity sensor.
+    /// </summary>
+    [Interface("AM2320 - Temperature and Humidity sensor")]
+    public class Am2320 : IDisposable
+    {
+        private readonly I2cDevice _i2c;
+        private readonly byte[] _readBuff = new byte[8];
+
+        /// <summary>
+        /// AM3220 default I2C address.
+        /// </summary>
+        public const int DefaultI2cAddress = 0x5C;
+
+        /// <summary>
+        /// The minimum read period is 1.5 seconds. Do not read the sensor more often.
+        /// </summary>
+        public static readonly TimeSpan MinimumReadPeriod = new TimeSpan(0, 0, 0, 1, 500);
+
+        private DateTime _lastMeasurement = DateTime.UtcNow.Subtract(MinimumReadPeriod);
+
+        /// <summary>
+        /// Gets a value indicating whether last read went, <c>true</c> for success, <c>false</c> for failure.
+        /// </summary>
+        public bool IsLastReadSuccessful { get; internal set; }
+
+        /// <summary>
+        /// Gets the last read temperature.
+        /// </summary>
+        /// <remarks>
+        /// If last read was not successful, it returns <code>default(Temperature).</code>
+        /// </remarks>
+        [Telemetry]
+        public Temperature Temperature
```

Comment:
Preferably change these properties to TryGetters. The operation is potentially expensive, has a side effect and may fail. We've had a bunch of problems with this `if (IsLastReadSuccessful())` approach for the DHT sensor family, which is conceptually very similar.

Reply:
That code was done before we adjusted all those. So adjusting for the Try pattern. I'll adjust that back to nanoFramework and will use the Obsolete as well on nanoFramework.